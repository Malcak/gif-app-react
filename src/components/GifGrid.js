import React from 'react';
import PropTypes from 'prop-types';
import { useFechGifs } from '../hooks/useFechGifs';
import { GifGridItem } from './GifGridItem';

export const GifGrid = ({ category }) => {
  const { data: images, loading } = useFechGifs(category);

  return (
    <>
      <h3>{category}</h3>
      {loading && <p>Loading...</p>}
      <div className="card-grid">
        {images.map((img) => (
          <GifGridItem key={img.id} {...img} />
        ))}
      </div>
    </>
  );
};

GifGrid.propTypes = {
  category: PropTypes.string.isRequired,
};
