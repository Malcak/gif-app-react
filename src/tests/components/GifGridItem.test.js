import React from 'react';
import { shallow } from 'enzyme';

import { GifGridItem } from '../../components/GifGridItem';

describe('tests on the <GifGridItem /> component', () => {
  const item = {
    id: 'VXJWhaO7afRe',
    title: 'One Punch Man GIF',
    url: '"https://media2.giphy.com/media/VXJWhaO7afRe/giphy.gif?cid=14cc3d561t7fzi4lu9a0uzxjbo3z5g7di6uhuhzc9go1k0b6&rid=giphy.gif&ct=g"',
  };

  let wrapper = shallow(<GifGridItem {...item} />);
  beforeEach(() => {
    wrapper = shallow(<GifGridItem {...item} />);
  });

  test('should display the <GifGridItem /> component correctly', () => {
    expect(wrapper).toMatchSnapshot();
  });
});
