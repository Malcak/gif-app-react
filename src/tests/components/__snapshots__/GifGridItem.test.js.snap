// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`tests on the <GifGridItem /> component should display the <GifGridItem /> component correctly 1`] = `
<div
  className="card animate__animated animate__fadeIn animate__delay-2s"
>
  <img
    alt="One Punch Man GIF"
    src="\\"https://media2.giphy.com/media/VXJWhaO7afRe/giphy.gif?cid=14cc3d561t7fzi4lu9a0uzxjbo3z5g7di6uhuhzc9go1k0b6&rid=giphy.gif&ct=g\\""
  />
  <p>
    title
  </p>
</div>
`;
